controlplane/kubeadm/internal/control_plane.go

```diff
@@ -238,6 +238,13 @@ func (c *ControlPlane) IsEtcdManaged() bool {
 	return c.KCP.Spec.KubeadmConfigSpec.ClusterConfiguration == nil || c.KCP.Spec.KubeadmConfigSpec.ClusterConfiguration.Etcd.External == nil
 }
 
+// UnhealthyMachinesWithNonMHCUnhealthyCondition returns all unhealthy control plane machines that
+// does not have a Kubernetes node or have any control plane component condition set to False.
+// It is different from the UnhealthyMachines func which checks MachineHealthCheck conditions.
+func (c *ControlPlane) UnhealthyMachinesWithNonMHCUnhealthyCondition(machines collections.Machines) collections.Machines {
```

Comment:
```suggestion
// UnhealthyControlPlaneOrNodeMissingMachines returns all unhealthy control plane machines that
// do not have a Kubernetes node, or have unhealthy control plane components.
//
// It differs from UnhealthyMachines which checks `MachineHealthCheck` conditions.
func (c *ControlPlane) UnhealthyControlPlaneOrNodeMissingMachines(machines collections.Machines) collections.Machines {
```

Something like that?

Reply:
All suggestion are fixed. cc @vincepri